packages/components/input-otp/src/input-otp.tsx

```diff
@@ -0,0 +1,95 @@
+import {forwardRef} from "@nextui-org/system";
+import {useMemo} from "react";
+
+import {UseInputOtpProps, useInputOtp} from "./use-input-otp";
+import {InputOtpSegment} from "./input-otp-segment";
+
+export interface InputOtpProps extends UseInputOtpProps {}
+
+const InputOtp = forwardRef<"div", InputOtpProps>((props, ref) => {
+  const {
+    Component,
+    otplength,
+    domRef,
+    styles,
+    value,
+    isInputFocused,
+    values,
+    hasHelper,
+    isInvalid,
+    errorMessage,
+    description,
+    getBaseProps,
+    getInputWrapperProps,
+    getInputProps,
+    getSegmentWrapperProps,
+    getHelperWrapperProps,
+    getErrorMessageProps,
+    getDescriptionProps,
+    ...otherProps
+  } = useInputOtp({
+    ...props,
+    ref,
+  });
+
+  const segmentsSection = useMemo(() => {
+    return (
+      <div {...getSegmentWrapperProps()}>
+        {Array.from(Array(otplength)).map((_, idx) => (
+          <InputOtpSegment
+            key={"segment-" + idx}
+            accessorIndex={idx}
+            classNames={values.classNames}
+            isInputFocused={isInputFocused}
+            otplength={otplength}
+            slots={values.slots}
+            value={value}
```

Comment:
suggest to use context inside `InputOtpSegment` instead of passing those props one by one in case there will be more props in the future. you may refer to popover component.